[TAN-171] Confirmation email for event registration

adessy opened the pull request:
# Changelog
## Added
- Send an email confirmation when a user registers for an event.

adessy committed:
[TAN-171] Add event urls to UrlService

adessy committed:
[TAN-171] Implement minimal version of confirmation email for event registrations

adessy committed:
[TAN-171] Implement the complete email for event registration confirmations

- Add missing event details
- Add functional buttons (Add to calendar / Go to event)
- Add missing event details

We didn't include the online link since it has not been released yet.

adessy committed:
[TAN-171] Add tests for EventRegistrationConfirmation

adessy committed:
[TAN-171] Add tests for EmailCampaigns::EventRegistrationConfirmationMailer

adessy committed:
[TAN-171] Implement mail preview for confirmation of email registrations

amanda-anderson commented:
@adessy Is this email finished? Or is there still some work to do? :)

adessy commented:
@amanda-anderson I thought it was pretty much done, but I could not make it work on epics yet (works just fine locally).

amanda-anderson commented:
@adessy Ok - do you think it will work on production though, as long as it's been working properly locally? :thinking: Maybe if we merged it we could test on staging quickly before releasing to production or something? :)

adessy committed:
[TAN-171] Add ICS attachment to confirmation emails for event registrations

adessy commented:
@amanda-anderson I had to fix a few things but now it's ready for review and translations.

adessy committed:
[TAN-171] Localize header message of confirmation emails for event registrations

adessy committed:
[TAN-171] Add missing final newlines

adessy committed:
[TAN-171] Simplify test environment configuration

Specifying `config.action_mailer.default_url_options` is no longer necessary because we now retrieve the host from the `AppConfiguration.instance` in `EmailCampaigns::EventRegistrationConfirmationMailer#add_to_calendar_url`.

adessy committed:
[TAN-171] Fix the image in confirmation emails

Include the image only if the image is available.

jamesspeake reviewed (approved):
All looks OK to me. Still think there are way too many files required to create a single email!
